integration/analytics_test.go

```diff
@@ -0,0 +1,174 @@
+//+build integration
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/windmilleng/wmclient/pkg/analytics"
+
+	"github.com/windmilleng/tilt/internal/testutils/tempdir"
+)
+
+const WindmillDirEnvVarName = "WINDMILL_DIR"
+const AnalyticsUrlEnvVarName = "TILT_ANALYTICS_URL"
+
+type analyticsFixture struct {
+	*k8sFixture
+	tempDir *tempdir.TempDirFixture
+	mss     *MemoryStatsServer
+}
+
+func newAnalyticsFixture(t *testing.T) *analyticsFixture {
+	td := tempdir.NewTempDirFixture(t)
+	af := &analyticsFixture{
+		k8sFixture: newK8sFixture(t, "analytics"),
+		tempDir:    td,
+	}
+	af.tiltEnviron[WindmillDirEnvVarName] = td.Path()
+
+	af.SetupAnalyticsServer()
+
+	return af
+}
+
+func (af *analyticsFixture) SetupAnalyticsServer() {
+	mss, port, err := StartMemoryStatsServer()
+	if !assert.NoError(af.t, err) {
+		af.t.FailNow()
+	}
+	af.mss = mss
+	delete(af.tiltEnviron, "TILT_DISABLE_ANALYTICS")
+	af.tiltEnviron[AnalyticsUrlEnvVarName] = fmt.Sprintf("http://localhost:%d/report", port)
+}
+
+func (af *analyticsFixture) TearDown() {
+	err := af.mss.TearDown()
+	if err != nil {
+		af.t.Fatal(err)
+	}
+	af.tempDir.TearDown()
+	af.k8sFixture.TearDown()
+}
+
+type envVarValue struct {
+	name  string
+	isSet bool
+	val   string
+}
+
+func saveEnvVar(name string) envVarValue {
+	val, isSet := os.LookupEnv(name)
+	return envVarValue{
+		name:  name,
+		isSet: isSet,
+		val:   val,
+	}
+}
+
+func restoreEnvVar(v envVarValue) error {
+	if !v.isSet {
+		return os.Unsetenv(v.name)
+	} else {
+		return os.Setenv(v.name, v.val)
+	}
+}
+
+func (af *analyticsFixture) SetOpt(opt analytics.Opt) {
+	v := saveEnvVar(WindmillDirEnvVarName)
+	err := os.Setenv(WindmillDirEnvVarName, af.tempDir.Path())
+	if err != nil {
+		af.t.Fatal(err)
+	}
+	err = analytics.SetOpt(opt)
+	if err != nil {
+		af.t.Fatal(err)
+	}
+	err = restoreEnvVar(v)
```

Comment:
wait a sec, if you restore the env var after you record the `choice.txt` file... how does tilt know where to look for the choice file when we `tilt up`?

Reply:
https://github.com/windmilleng/tilt/blob/241f7a83d70483b6178b225dbb19f39dcc872a64/integration/analytics_test.go#L33

Reply:
👍